addons/l10n_pk_edi/models/product_product.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    l10n_pk_edi_sale_type = fields.Many2one('l10n.pk.product.type', 'Sale Type',
+        compute='_compute_l10n_pk_edi_sale_type', required=True, readonly=False, store=True,
+    )
+    l10n_pk_edi_schedule_code = fields.Many2one('l10n.pk.schedule.code', 'Schedule Code',
+        compute='_compute_l10n_pk_edi_schedule_code', required=True, readonly=False, store=True,
+    )
+
+    @api.depends('type')
+    def _compute_l10n_pk_edi_sale_type(self):
+        def _get_pk_sale_type(type):
+            match type:
+                case 'consu':
+                    return 'T1000017'
+                case 'service':
+                    return 'T1000018'
+                case _:
+                    # Otherwise, default to 'T1000083' which represent 'Other'
+                    return 'T1000083'
```

Comment:
Wait wait 😅 

- do not override builtin types like `type` with args
- dict.get uses 1 instruction, instead of multiple comparisons.
  in cases like this it's just simpler.

```suggestion
        def _get_pk_sale_type(sale_type):
            return {
               'consu': 'T1000017',
               'service': 'T1000018',
            }.get(sale_type, 'T1000083')
```